components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java

```diff
@@ -253,14 +254,32 @@ private void revokeTokensOfBindingRef(AuthenticatedUser user, String tokenBindin
         if (StringUtils.isBlank(tokenBindingReference) || user == null) {
             return;
         }
+        String userId;
+        try {
+            userId = user.getUserId();
+        } catch (UserIdNotFoundException e) {
+            log.error("User id cannot be found for user: " + user.getLoggableUserId() + ". Hence skip revoking " +
+                    "relevant tokens");
+            throw new IdentityOAuth2Exception("Unable to revoke tokens for the token binding reference: "
+                    + tokenBindingReference);
+        }
+
         Set<AccessTokenDO> boundTokens = OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
                 .getAccessTokensByBindingRef(tokenBindingReference);
         for (AccessTokenDO accessTokenDO : boundTokens) {
             String consumerKey = accessTokenDO.getConsumerKey();
-            if (OAuth2Util.getAppInformationByClientId(consumerKey)
-                    .isTokenRevocationWithIDPSessionTerminationEnabled() && accessTokenDO.getAuthzUser() != null &&
-                    user.toString().equals(accessTokenDO.getAuthzUser().toString())) {
-                revokeTokens(consumerKey, accessTokenDO, tokenBindingReference);
+            if (OAuth2Util.getAppInformationByClientId(consumerKey).isTokenRevocationWithIDPSessionTerminationEnabled()
```

Comment:
Did we test this fix for JIT provisioned users?

Reply:
Yes. tested for JIT provisioned business users with sso-session binding enabled